### ISSUE TITLE
MongoDB metricsets incorrectly handling multiple hosts

### ISSUE THREAD

**Issue by jsoriano** [elastic/beats#32188]
Metricsets of the MongoDB module are doing their own parsing of the `hosts` setting ([here](https://github.com/elastic/beats/blob/57c237194a6dabccb5d1545114e35a628071119b/metricbeat/module/mongodb/mongodb.go#L80)), and using the result to configure the MongoDB client ([here](https://github.com/elastic/beats/blob/57c237194a6dabccb5d1545114e35a628071119b/metricbeat/module/mongodb/mongodb.go#L157)).

Usually, hosts parsing is done by the module, and individual metricsets should use the host data in the base metric set. This was also being done in this module at least till 8.2 ([here](https://github.com/elastic/beats/blob/0279ca20dc1f10a34c7497103c52bea76dfd74c2/metricbeat/module/mongodb/metricset.go#L50)).

The new approach produces failures when multiple hosts are used:
```
fetching data for metricset mongodb.dbstats: could not create mongodb client: could not create mongodb client: a direct connection cannot be made if multiple hosts are specified
```

For confirmed bugs, please report:
- Version: 8.3
- Operating System: Probably all.
- Discuss Forum URL: https://discuss.elastic.co/t/failed-to-monitor-mongo-cluster/308453
- Steps to Reproduce: Use the MongoDB Metricbeat module with multiple hosts.

**Comment by ritalwar** [comment]
The issue is fixed in PR : https://github.com/elastic/beats/pull/34624.

**Comment by ritalwar** [comment]
During testing, some issues were identified with the fix. As a result, I have made modifications the fix. Currently, the fix is undergoing testing and review.

**Comment by rameshelastic** [comment]
Can we elaborate the issues identified @ritalwar ?

**Comment by ritalwar** [comment]
I encountered a situation where I developed a code fix and performed thorough testing in my environment. Everything appeared to be functioning correctly during testing in my setup. However, when @shmsr  tested the fix, he discovered an issue. The connection was erroneously falling back to the default host and port, despite the configuration specifying a different host and port.

Despite my efforts, I was unable to replicate this issue in my environment. As a troubleshooting measure, I attempted the fix on a different machine, where I successfully replicated the issue. Upon further investigation and debugging, I identified the root cause: the client options were being initialized twice in the code, resulting in the overwritten URI that should have been applied.

To address this, I have rectified the issue by ensuring that the client options are initialized only once, preventing the overwrite of the intended URI.
